cmd/shortener/main.go

```diff
@@ -5,26 +5,41 @@ import (
 	"go-shortener/internal/app"
 	"go-shortener/internal/config"
 	"go-shortener/internal/handlers"
+	"go-shortener/internal/storage"
 	"log"
 	"net/http"
 )
 
-func NewRouter() chi.Router {
+func NewRouter(app handlers.Application, baseURL string) chi.Router {
 	r := chi.NewRouter()
+	h := handlers.NewShortenHandlers(app)
+	h.SetBaseURL(baseURL)
 
-	r.Get("/{ID:[a-zA-Z0-9]+}", handlers.GetURLHandler)
-	r.Post("/", handlers.CreateHandler)
-	r.Post("/api/shorten", handlers.CreateJSONHandler)
+	r.Get("/{ID:[a-zA-Z0-9]+}", h.GetURLHandler)
+	r.Post("/", h.CreateHandler)
+	r.Post("/api/shorten", h.CreateJSONHandler)
 	return r
 }
 
 func main() {
-	config.Init()
-	err := app.Init()
+	cnf, err := config.Init()
 	if err != nil {
 		log.Fatal(err)
 	}
-	err = http.ListenAndServe(config.Config.ServerAddress, NewRouter())
+
+	var st app.Storage
+	if cnf.FileStorage != "" {
+		st, err = storage.NewFileStorage(cnf.FileStorage)
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		st = storage.NewMemoryStorage()
+	}
```

Comment:
небольшой трюк, если вдруг понравится:
```suggestion
	if cnf.FileStorage != "" {
		st, err = storage.NewFileStorage(cnf.FileStorage)
	} else {
		st = storage.NewMemoryStorage()
	}
	if err != nil {
		log.Fatal(err)
	}
```
как то так, если вдруг захочется вложенность избежать :)

Reply:
Спасибо 🙂